spec/widgets/category/category-widget-model.spec.js

```diff
@@ -32,6 +33,23 @@ describe('widgets/category/category-widget-model', function () {
     });
   });
 
+  describe('search state', function () {
```

Comment:
Yes, the repetitive `this.dataviewModel.*` usages is a [feature envy](https://sourcemaking.com/refactoring/smells/feature-envy) smell. The only side-effect that seems relevant for DI/the widget is to update the `selected` based on the search result change, somehow (based on the code snippet here).